src/api/app/models/concerns/staging_project.rb

```diff
@@ -142,35 +142,22 @@ def cache_problems
   end
 
   def state
-    return :empty unless staged_requests.exists?
-    return :unacceptable if unacceptable_state?
-    return :review if review_state?
-    general_state
-  end
-
-  def unacceptable_state?
-    untracked_requests.present? || staged_requests.obsolete.present?
-  end
-
-  def review_state?
-    general_state == :acceptable && missing_reviews.present?
-  end
-
-  def general_state
-    build_state == :acceptable ? check_state : build_state
+    return :empty if staged_requests.blank?
+    return :unacceptable if untracked_requests.present? || staged_requests.obsolete.exists?
```

Comment:
I'm wondering if now is better than before... Is that because of code climate? IMO The code climate cognitive complexity is definitely higher than before (introduced by the or condition). I would prefer to have it was before

Reply:
When we read this code on the mob-programming session, I realized most of us got confused because we had to jump to a different method on each condition. And even more tricky was that we called `general_state` at the end on the main method `state` but also from `review_state?` method. So the methods called each other :confounded: 

I really think it is clearer now, not so many confusing jumps.

But anyway, we can still improve it. And yes, code climate restrict it a lot... read the following comments.